src/Network-Tests/UUIDGeneratorTest.class.st

```diff
@@ -18,67 +18,75 @@ UUIDGeneratorTest >> setUp [
 
 { #category : #testing }
 UUIDGeneratorTest >> testCounterRollover [
+
 	| previousUuid uuid |
-	previousUuid := generator next.
+	previousUuid := ByteArray new: 16.
+	UUIDGenerator default placeFields: previousUuid.
 	"the internal counter inside generator will rollover at 16r10000,
 	the least significant 10th byte rolls over every 16r100,
 	the internal counter starts at a random value"
 	2 * 16r10000 timesRepeat: [
-		uuid := generator next.
+		uuid := ByteArray new: 16.
+		UUIDGenerator default placeFields: uuid.
 		"same generator, consequetive counters (take roll-over into account)"
-		self assert: ((previousUuid at: 10) + 1) \\ 256 equals: (uuid at: 10).
+		self assert: (previousUuid at: 10) + 1 \\ 256 equals: (uuid at: 10).
 		previousUuid := uuid ]
 ]
 
 { #category : #testing }
 UUIDGeneratorTest >> testDefault [
-	self assert: UUIDGenerator default equals: UUIDGenerator default.
-	self deny: UUIDGenerator next isNilUUID
-]
 
-{ #category : #testing }
-UUIDGeneratorTest >> testOne [
-	| uuid |
-	uuid := generator next.
-	self assert: (uuid isKindOf: UUID).
-	self deny: uuid isNilUUID
+	self assert: UUIDGenerator default equals: UUIDGenerator default.
 ]
 
 { #category : #testing }
 UUIDGeneratorTest >> testSpeed [
+	|array|
+	array := ByteArray new: 16.
 	self
-		should: [ 100000 timesRepeat: [ generator next ] ]
+		should: [ 100000 timesRepeat: [ UUIDGenerator default placeFields: array ] ]
 		notTakeMoreThan: 1 second
 ]
 
 { #category : #testing }
 UUIDGeneratorTest >> testTwoDifferentGenerator [
+
 	| generator1 generator2 uuid1 uuid2 time1 time2 |
 	generator1 := UUIDGenerator new.
 	generator2 := UUIDGenerator new.
-	uuid1 := generator1 next.
-	uuid2 := generator2 next.
+	uuid1 := NilUUID new.
+	uuid2 := NilUUID new.
+	generator1 placeFields: uuid1 uuidData.
+	generator2 placeFields: uuid2 uuidData.
```

Comment:
This test breaks the public interface. We should kill it probably or rewrite it :)

Let's discuss

Reply:
NilUUID has been removed :)